mwdb/core/search/parse_helpers.py

```diff
@@ -0,0 +1,402 @@
+    "string" nodes are unescaped.
+    """
+    # Ensure that \ is not a control character because it makes things
+    # more complicated. Lucene doesn't allow for unescaped \ anyway
+    assert "\\" not in control_chars
+    if control_chars:
+        # Escape regex control characters in set of control characters
+        # to be safely put in square brackets [...]
+        re_control_chars = re.sub(r"([\^\-\]])", r"\\\1", control_chars)
+        pattern = rf"[{re_control_chars}]|\\."
+    else:
+        pattern = r"\\."
+
+    current_pos = 0
+    current_string = ""
+
+    for match in re.finditer(pattern, value):
+        start, end = match.span(0)
+        if current_pos < start:
+            current_string += value[current_pos:start]
+        if value[start] == "\\":
+            escaped_char = value[start + 1]
+            if escaped_char in control_escapes:
+                if current_string:
+                    yield TokenType.STRING, current_string
+                    current_string = ""
+                yield TokenType.CONTROL, value[start:end]
+            else:
+                current_string += value[start + 1 : end]
+        else:
+            if current_string:
+                yield TokenType.STRING, current_string
+                current_string = ""
+            yield TokenType.CONTROL, value[start:end]
+        current_pos = end
+
+    if current_pos < len(value):
+        current_string += value[current_pos:]
+    if current_string:
+        yield TokenType.STRING, current_string
+
+
+def join_tokenized_string(tokenized_string: Iterable[StringToken]) -> str:
+    return "".join(s for _, s in tokenized_string)
+
+
+def split_tokenized_string(
+    tokenized_string: List[StringToken], separator: StringToken
+) -> List[List[StringToken]]:
+    last_index = 0
+    for sep_index in [
+        idx for idx, token in enumerate(tokenized_string) if token == separator
+    ]:
+        yield tokenized_string[last_index:sep_index]
+        last_index = sep_index + 1
+    yield tokenized_string[last_index : len(tokenized_string)]
+
+
+def parse_field_path(field_path: str) -> PathSelector:
+    """
+    Extract subfields from fields path with proper control character handling:
+
+    - \\x - escaped character
+    - * - array element reference e.g. (array*:2)
+    - . - field separator
+    """
+    path_selector: PathSelector = []
+    tokenized_string = tokenize_string(field_path, "*.")
+    path_elements = split_tokenized_string(
+        list(tokenized_string), (TokenType.CONTROL, ".")
+    )
+    for path_element in path_elements:
+        asterisk_count = 0
+        for asterisk_count, token in enumerate(reversed(path_element)):
+            if token != (TokenType.CONTROL, "*"):
+                break
+        element_name = (
+            path_element[:-asterisk_count] if asterisk_count > 0 else path_element
+        )
+        path_selector.append((join_tokenized_string(element_name), asterisk_count))
+    return path_selector
```

Comment:
I would appreciate more comments in this section. Something like in `make_jsonpath_selector()`